### PR TITLE
For Mereg : Added authtoken with embed URL when user login via SSO and open the page with forum is embedded. (Ticket id: 51627)

### PR DESCRIPTION
**Pull request by kishanm-wtbx**
For Merge

Added authtoken with embed URL when user login via SSO and open the page with forum is embedded. 
Fixed the mantis [0021079: Unable to logout from the forum](https://www.websitetoolbox.com/mantis/view.php?id=21079)

Ticket: [https://www.websitetoolbox.com/tool/ticket/view/51627/qP1tTkiMA7Fp50td5eYw](https://www.websitetoolbox.com/tool/ticket/view/51627/qP1tTkiMA7Fp50td5eYw)

QA: Gopal Kumar

Note: Require to merge into craftcms_v4 branch. After merge I will create a git tag and bind tag with latest commit. so, craftcms store will fetch latest plugin using tag.

### CODE REVIEW

File: src/services/Sso.php [hunk 1]
```diff
@@ -172,32 +172,40 @@ function resetCookieOnLogout(){
...
           { var embedUrl  = "{$forumUrl}";
             var userStatus = "{$userStatus}";
+            var authToken = "{$token}"; 
             var wtbWrap = document.createElement('div');
             wtbWrap.id = "wtEmbedCode";            
             var embedScript = document.createElement('script');
             embedScript.id = "embedded_forum";
             embedScript.type = 'text/javascript'; 
-            embedUrl += "/js/mb/embed.js";
+            embedUrl += "/js/mb/embed.js?"+authToken
```
Comment (masihur) [inline src/services/Sso.php]: Add a condition like as below.
embedUrl += "/js/mb/embed.js;
if(isset(authtoken) && authtoken != "") {
 embedUrl += authToken;
}

File: src/services/Sso.php [hunk 2]
```diff
@@ -172,32 +172,40 @@ function resetCookieOnLogout(){
...
+        $token = '';
+        if((isset($_COOKIE['forumLogInToken']) && $_COOKIE['forumLogInToken'] != '')|| ($domainChange && isset($_COOKIE['forumLogoutToken']))){
+            $cookieForumLoginToken = isset($_COOKIE['forumLogInToken'])?$_COOKIE['forumLogInToken'] : $_COOKIE['forumLogoutToken'];
+            setcookie("forumLogInToken", '', time() - (86400 * 365),"/"); 
+            $_COOKIE['forumLogInToken'] = '';
+            Craft::$app->getProjectConfig()->remove('plugins.websitetoolboxforum.settings.isDomainChange');
+            $token = '?authtoken='.$cookieForumLoginToken;
+            echo '<img src='.$forumUrl.'/register/dologin?authtoken='.$cookieForumLoginToken.'  width="0" height="0" border="0" alt="">';
+        }
+        if($userStatus == 'loggedout'){
```
Comment (masihur) [inline src/services/Sso.php]: No need to use it.

**Review by masihur — CHANGES_REQUESTED** [review]

**Review by masihur — APPROVED** [review]

File: src/services/Sso.php [hunk 3]
```diff
@@ -172,32 +172,38 @@ function resetCookieOnLogout(){
...
             embedUrl += "/js/mb/embed.js";
+            if(authToken != ''){ embedUrl += authToken; }
             embedScript.src = embedUrl;
-            embedScript.setAttribute('data-version','1.1');            
+            embedScript.setAttribute('data-version','1.1');
             wtbWrap.appendChild(embedScript);
             if(document.getElementById('wtEmbedCode') != null){
                 document.getElementById('wtEmbedCode').innerHTML = '';
-                document.getElementById('wtEmbedCode').appendChild(embedScript);  
+                setTimeout(function(){
```
Comment (masihur) [inline src/services/Sso.php]: Why you use timeout. I think no need to use timeout

Comment (webtoolbox) [inline src/services/Sso.php]: Why do you need the timeout? Why can't you do this with an event handler instead of a timeout?

**Review by webtoolbox — CHANGES_REQUESTED** [review]
Your pull request description should say what the issue was and how you fixed it in non-technical terms. Please update it.

File: src/services/Sso.php (hunk 3 above)
Comment (masihur) [reply src/services/Sso.php]: I already update it to Kishan, no need to use timeout. He is updating the code.

**Review by masihur — APPROVED** [review]

**Comment by webtoolbox** [comment]
Then I don't understand why you approved the pull request and asked for my
review.

Thank you,
Sandeep Sondhi
Chief Executive Officer
Website Toolbox
www.WebsiteToolbox.com <https://www.websitetoolbox.com>

**Review by webtoolbox — APPROVED** [review]

**Review by webtoolbox — APPROVED** [review]

**Pull request merged**